src/index.ts

```diff
@@ -0,0 +1,54 @@
+import { fetchVoteData } from "./queries/voting-data";
+const Web3 = require("web3");
+require("dotenv").config();
+const fs = require("fs");
+
+if(!process.env.VOTE_ID) {
+  throw Error("Vote ID is required as an env variable")
+}
+
+// This is the vote we are distributing rewards for
+const voteId = process.env.VOTE_ID
+
+// TODO: This will come from the collections contract
+const DISTRIBUTION_AMOUNT = 10000
+
+const web3 = new Web3(
+  new Web3.providers.HttpProvider(
+    `https://${process.env.ETHEREUM_NETWORK}.infura.io/v3/${process.env.INFURA_API_KEY}`
+  )
+);
+
+const stakingContractABI  = JSON.parse(fs.readFileSync("src/abi/sarcoStaking.json"));
+const stakingContract = new web3.eth.Contract(
+  stakingContractABI,
+  process.env.CONTRACT_ADDRESS
+);
+
+async function main() {
+  const voteData = await fetchVoteData(web3, voteId)
+  const blockNumber = voteData.snapshotBlockNumber
+ 
+  const didVoteAddresses = new Map<string, number>()
+  let totalVoteBalance = 0
+  for (let i = 0; i < voteData.addresses.length; i++) {
+    const votingAddress = voteData.addresses[i]
+    const stakedValueAt = await stakingContract.methods.stakeValueAt(votingAddress,blockNumber).call()
+    totalVoteBalance += +stakedValueAt
```

Comment:
One question I have here is whether we need to use BigNumbers here (`web3.utils.BN`). For the example that is setup this runs well, but if the collections contract returns a number with 18 decimals, we may run into issues with the number formatting.

Reply:
100% need to use BigNumbers here. I'm surprised this code even works as is, I'd expect `stakedValueAt` to already be a `BN` object (which means the operation on line 37 shouldn't work, because it's not a primitive `Number`) . Maybe the `+` in `+stakedValueAt` converts it to a `Number`?

Reply:
Yeah, I had to look that one up -- it does convert it to a number. We will be using BigNumbers here, have added as a task to the backlog.